src/Layers/xrRender/r__dsgraph_render.cpp

```diff
@@ -593,6 +594,31 @@ void D3DXRenderBase::r_dsgraph_render_subspace(IRender_Sector* _sector, CFrustum
                 renderable->renderable_Render(renderable);
             }
         }
+#if RENDER != R_R1
+        if (g_pGameLevel && (phase == RImplementation.PHASE_SMAP) && ps_r__common_flags.test(RFLAG_ACTOR_SHADOW)) // Actor Shadow (Sun + Light)
+        {
+            do {
+                IGameObject* viewEntity = g_pGameLevel->CurrentViewEntity();
+                if (viewEntity == nullptr)
+                    break;
+                viewEntity->spatial_updatesector();
+                CSector* sector = (CSector*)viewEntity->GetSpatialData().sector;
+                if (nullptr == sector)
+                    break; // disassociated from S/P structure
+                if (PortalTraverser.i_marker != sector->r_marker)
+                    break; // inactive (untouched) sector
+                for (u32 v_it = 0; v_it < sector->r_frustums.size(); v_it++)
```

Comment:
Consider to use range-based for:
```suggestion
                for (const CFrustum& view : sector->r_frustums)
```